src/main/scala/abalone/variant/Variant.scala

```diff
@@ -102,6 +103,144 @@ abstract class Variant private[variant] (
         )
     }.toMap
 
+  def validMovesOf2(situation: Situation, validMoves1: Map[Pos, List[Move]]): Map[Pos, List[(String, Move)]] =
+    situation.board.piecesOf(situation.player).flatMap {
+      case ((pos, piece)) => {
+        Map(
+          pos -> pos.neighbours.flatMap {
+            case Some(neighbour) if(situation.board.piecesOf(situation.player).contains(neighbour)) => {
+              neighbour.neighbours.flatMap {
+                case Some(neighbourOfNeighbour)
+                  if (situation.board.isEmptySquare(Some(neighbourOfNeighbour))
+                  && (pos.isInLine(neighbour, neighbourOfNeighbour) || situation.board.isEmptySquare(pos.dir(neighbour.dir(neighbourOfNeighbour))))
+                  && !validMoves1.get(pos).toList.flatMap(_.map(_.toUci)).exists(m => m.toString == s"Move(${pos},${neighbourOfNeighbour},None)")) =>
```

Comment:
i think there must be a way not to need to look at `validMoves1` here. Surely by checking the whole neighbour of neighbours we are doing a search that involves three `Pos` here, and so we can't be generating a move of 1 in this code?

Reply:
you could probably filter our moves where the origin and dest are neighbours as no moves of two can have that form?

Reply:
yes, i don't think there is a need to have to check `validMovesOf1` in `validMovesOf2` and it would be much more efficient if we didn't have to pass in a parameter `validMovesOf1`.